hibernate-core/src/test/java/org/hibernate/test/hql/HQLTest.java

```diff
@@ -1538,4 +1501,11 @@ public void testComponentNoAlias() throws Exception {
 		compileWithAstQueryTranslator( "from Human where name.first = 'Gavin'", false);
 	}
 
+	@Test
+	public void testCountDistinctOfSubquery(){
+		createNewQueryTranslator(
+				"select count (DISTINCT (SELECT function('lower', 'DAY') from Bar)) from Bar"
```

Comment:
Actually SQL-92 does explicitly disallow this, but I am not sure if a newer standard allows for it.
http://www.contrib.andrew.cmu.edu/~shadow/sql/sql1992.txt see 6.5  <set function specification>